examples/idempotent_producer_example/idempotent_producer_example.go

```diff
@@ -0,0 +1,178 @@
+// Idempotent Producer example.
+//
+// The idempotent producer provides strict ordering and
+// exactly-once producing guarantees.
+//
+// From the application developer's perspective, the only difference
+// from a standard producer is the enabling of the feature by setting
+// the `enable.idempotence` configuration property to true, and
+// handling fatal fatal errors (Error.IsFatal()) which are raised when the
```

Comment:
a fatal fatal error sounds really bad.

Reply:
These things should not be underestimated, I'll add a third fatal for clarity